README.md

```diff
@@ -1,70 +1,363 @@
+    make -C libyui-bindings -f Makefile.repo build
+
+
+or explicitly select subprojects to build with or without like
+`-p` (`--no-pkg`) or `-g` (`--no-graph`) or `-r` (`--rest-api`).
+See `./build-all -h` for an up-to-date complete list.
+
+
+### Daily Development Work: Keeping an Existing Build
+
+When working on libyui, it is common to change or add a class on the abstract
+libyui level, then extend the concrete implementation in the Qt and NCurses
+UIs; you don't want to rebuild everything from scratch after every little
+change.
+
+If you use `build-all build`, it will do just that: It will remove each
+`build/` subdirectory in each subproject, recreate it, invoke `cmake`, then
+`make`.
 
+If you simply call it without any make target, however, it will check in each
+subproject if there is an existing `build/` subdirectory, and if there is, just
+call `make`. If there is no `build/` subdirectory, it will call `make -f
+Makefile.repo build` instead. Notice that this handles each subproject
+individually, so you can have a mixture of existing and non-existing `build/`
+subdirectories.
+
+    ./build-all
+
+This builds everything from scratch.
+
+Now continue editing files in subprojects and then
+
+    ./build-all
+
+This now just invokes `make` in each subproject, keeping the object
+files for unchanged sources.
+
+### make install
+
+Of course you need root permissions to install any file to the system:
+
+    ./build-all
+    sudo ./build-all install
+
+
+### Keeping Changes Locally in the Source Tree
+
+The CMake environments in the subprojects are set up to prefer header files and
+built libraries from sibling subprojects over those from the system; so you can
+work for a long time in the source tree without a need for `sudo make install`.
+
+For the slightly paranoid:
+
+To make absolutely sure that you are only using headers and libs from your
+source tree, you can move the headers away in the system directories while you
+develop and later move them back:
+
+    cd /usr/include
+    sudo mv yui yui.orig
+
+    cd /usr/lib64
+    sudo mkdir libyui.save
+    sudo mv libyui.so* libyui.save
+    sudo mv yui libyui.save
+
+Make sure to restore this later when you are done:
+
+    cd /usr/include
+    sudo rm -rf yui
+    sudo mv yui.orig yui
+
+    cd /usr/lib64
+    sudo rm -rf libyui.so* yui
+    sudo mv libyui.save/* .
+    sudo rmdir libyui.save
+
+_If you wrecked it, force-reinstall the libyui packages:_
+
+    rpm -qa "libyui*" >/tmp/libyui-pkg
+    sudo zypper install --force $(cat /tmp/libyui-pkg)
+
+
+# Version Numbers
+
+All packages in this source tree have the same version numer; that's why the
+subprojects share the toplevel `VERSION.cmake` file.
+
+The general idea is to enable transaction-like changes on the libyui base
+packages to avoid long delays in the distribution build cycle and broken builds:
+
+Some parts may have become incompatible, yet the higher-level parts require the
```

Comment:
I don't want to explain the whole wide world of OBS here to those who don't know it at all. Don't forget that a good portion of people using and reading this will be from SUSE R&D, and that should be enough explanations to avoid repeating the same questions in the future.